core/embed/rust/src/micropython/list.rs

```diff
@@ -2,25 +2,27 @@ use core::convert::TryFrom;
 
 use crate::error::Error;
 
-use super::{ffi, gc::Gc, obj::Obj};
+use super::{ffi, gc::Gc, obj::Obj, runtime::catch_exception};
 
 pub type List = ffi::mp_obj_list_t;
 
 impl List {
-    pub fn alloc(values: &[Obj]) -> Gc<Self> {
+    pub fn alloc(values: &[Obj]) -> Result<Gc<Self>, Error> {
         // SAFETY: Although `values` are copied into the new list and not mutated,
         // `mp_obj_new_list` is taking them through a mut pointer.
-        unsafe {
+        // EXCEPTION: Will raise if allocation fails.
+        catch_exception(|| unsafe {
             let list = ffi::mp_obj_new_list(values.len(), values.as_ptr() as *mut Obj);
             Gc::from_raw(list.as_ptr().cast())
-        }
+        })
     }
 
-    pub fn append(&mut self, value: Obj) {
+    pub fn append(&mut self, value: Obj) -> Result<(), Error> {
         unsafe {
             let ptr = self as *mut Self;
             let list = Obj::from_ptr(ptr.cast());
-            ffi::mp_obj_list_append(list, value);
+            // EXCEPTION: Will raise if allocation fails.
+            catch_exception(|| ffi::mp_obj_list_append(list, value)).map(|_| ())
```

Comment:
Why the `map`?

Reply:
More generally, if we don't need the return value, I'd prefer throwing it away *in* the closure (`;` is an operator for that), and not only here, but in other uses of `catch_exception` as well.

Reply:
fixed along with one other case in b406bd0a0ff986392c2fccef336a7b9201734023